### PR TITLE
refactor: refactor ws adapter to reduce its cognitive complexity

### LINKED ISSUE #396: Refactor ws adapter to reduce its cognitive complexity
#### Reason/Context
the [_connect()](https://github.com/asyncapi/glee/blob/master/src/adapters/ws/server.ts#L23) method in `ws` server adapter currently has a cognitive complexity of 35. It's really hard to understand what is going on. plus, SONAR always complains about it in PRs 😆 

#### Description
refactor the _connect() method by grouping it's logic in separate functions for the purpose of readability and maintainability.

### PR DESCRIPTION
**Pull request by oviecodes**
**Description**
Refactor ws adapter _connect() method to reduce its cognitive complexity

**Related issue(s)**
fixes #396

### CODE REVIEW

**Review by KhudaDad414 — DISMISSED** [review]
LGTM! let's wait for @Souvikns review. :)

**Review by Souvikns — APPROVED** [review]
Looks good, @oviecodes you have some conflicts, you can merge this PR after you resolve them.

File: package-lock.json
Comment (KhudaDad414) [inline package-lock.json:1]: why is package-lock.json is being regenerated? you haven't touched package.json. please revert this.

Comment (oviecodes) [reply package-lock.json:1]: @KhudaDad414 I have reverted it.

**Comment by Souvikns** [comment]
@oviecodes can you resolve the conflicts, then we can move ahead with the review.

**Comment by oviecodes** [comment]
@Souvikns i've resolved the changes

**Review by KhudaDad414 — APPROVED** [review]
LGTM

**Pull request merged**
